### ISSUE TITLE
channel.presence.get() v1.2.18 throws PromiseRejectCallback

### ISSUE THREAD

**Issue by seekayel** [ably/ably-js#943]
New version (1.2.18) causes exception when checking channel presence. Rolling back to (1.2.17) resolves failure.

Don't have exact replication code yet, but here is some pseudo-code that is triggering the error.

```javascript
var Ably = require('ably/promises');
let client = Ably.Rest(process.env.ABLY_KEY)
let channel = await client.channels.get(channel_name)
let presence = channel.presence.get()
```

Error output:

```
Exception in PromiseRejectCallback:
node_modules/ably/browser/static/ably-node.js:597
return new Promise(function (resolve, reject) {
^
RangeError: Maximum call stack size exceeded
```

Any tips/pointers on how to debug/reproduce welcome.

┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-1713) by [Unito](https://www.unito.io)

**Comment by owenpearson** [comment]
Hey @seekayel thanks a lot for reporting this, that's very helpful. I can confirm this is a regression introduced by `1.2.18` and rolling back to `1.2.17` will fix this. I have found the root cause and it should be an easy fix so we'll release a new patch version with the fix in the near future.

I'll leave a reproduction script below for visibility:
```ts
const Ably = require('ably');

const key = '<YOUR_ABLY_API_KEY>';
const client = new Ably.Rest.Promise({ key });
const channel = client.channels.get('my_channel');

channel.presence.get().then(result => {
  // RangeError will throw before this code is reached
  console.log(result);
});
```

Btw, `client.channels.get` is a synchronous method so no need to `await` it 🙂

**Comment by seekayel** [comment]
@owenpearson good deal!

Did the CICD verification tests fail after merge?

Search for: presenceEnterWithData https://github.com/ably/ably-js/runs/5773616514?check_suite_focus=true

**Comment by owenpearson** [comment]
@seekayel The tests are currently passing on main (although a few of the tests are somewhat flakey). The root cause of this bug is specifically to do with how the library automatically generates promise-based versions of async methods (the actual implementations use callbacks instead of promises for browser compatibility). We should have caught this in CI but unfortunately we don't currently have full test coverage for the automatically generated promise-based methods. I've opened #944 to address this.

**Comment by owenpearson** [comment]
Hey @seekayel, I've just released 1.2.19 which includes a fix for this 🙂

**Comment by seekayel** [comment]
@owenpearson thanks. We pinned v1.2.17 to fix. Will try next after current feature.
